### DISCUSSION TITLE
[BUG?] ActiveSelection with preserveObjectStacking

### DISCUSSION THREAD

**Discussion by 0ro** [fabricjs/fabric.js#9193] [Q&A]
I'm trying a bit a last fabricjs beta 6.0.0-beta12. I want to programmatically select objects on Canvas. As far as I understand I can do it with creating ActiveSelection. But with `preserveObjectStacking: true` it works weirdly for me, and probably it is a bug. When you mouse over red areas on the screenshot inside the ActiveSelection group, the cursor doesn't understand that there is a group below, and if you click the selection is disappearing

![image](https://github.com/fabricjs/fabric.js/assets/7969190/0e6c2fd1-9720-494d-869f-455474d22643)

Is it expected behavior? Should I create a bug about it? I'm just not sure, it is still a beta, and maybe it will fixed later. I would like to help with it. Thanks

Here is my [codesandbox](https://codesandbox.io/s/fabricjs-playground-n4ngc?file=/src/index.js) playground

**Comment by 0ro** [comment]
By the way, the same logic throws an error on the 5.3.0 version.

`Cannot read properties of undefined (reading 'getRetinaScaling')`

Here [codesandbox with 5.3.0](https://codesandbox.io/s/fabricjs-playground-forked-wfdz3j?file=/src/index.js)

**Comment by ShaMan123** [comment]
Before anything read #7670

**Reply by 0ro** [reply]
Thank you, I've read, but sorry, still don't understand, how it should be fixed. Because if I manually, via mouse, make a selection of objects, it behaves normally, and I can drag the whole "group" by clicking at any place inside. But if I do it by creating ActiveSelection it works differently. Can I achieve the same behavior as I do via mouse selection?
